build.gradle

```diff
@@ -104,7 +132,309 @@ subprojects { project ->
+                def ossPass = System.getenv("SONATYPE_PASSWORD") ?: project.hasProperty("sonatypeOssPassword") ? project.sonatypeOssPassword : ''
+                def ossStagingProfileId = System.getenv("SONATYPE_STAGING_PROFILE_ID") ?: project.hasProperty("sonatypeOssStagingProfileId") ? project.sonatypeOssStagingProfileId : ''
+                nexusUrl = uri("https://s01.oss.sonatype.org/service/local/")
+                username = ossUser
+                password = ossPass
+                stagingProfileId = ossStagingProfileId
+            }
+        }
+        transitionCheckOptions {
+            maxRetries.set(40)
+            delayBetween.set(java.time.Duration.ofMillis(2000))
+        }
+    }
+}
+
+def DOCUMENTATION_GROUP = 'Documentation'
+
+def commonGithubOrg = 'grails'
+def commonGithubSlug = 'grails-common-build'
+def commonBranch = 'master'
+def docResourcesDir = "${buildDir}/resources/${commonGithubSlug}-${commonBranch}/src/main/resources"
+
+task prepareDocResources {
+    group = DOCUMENTATION_GROUP
+    description = 'Downloads common documentation resoruces and unzips them to build folder'
+    doLast {
+        ant.mkdir(dir: buildDir)
+        ant.get(src: "https://github.com/${commonGithubOrg}/${commonGithubSlug}/archive/${commonBranch}.zip", dest: "${buildDir}/resources.zip")
+        ant.unzip(src: "${buildDir}/resources.zip", dest: "${buildDir}/resources")
+    }
+}
+
+task copyLocalDocResources(type: Copy, dependsOn: prepareDocResources) {
+    group = DOCUMENTATION_GROUP
+    description = 'Copy local resources to build folder'
+    mustRunAfter prepareDocResources
+    from('src/main/docs/resources')
+    into docResourcesDir
+}
+
+configurations {
+    documentation
+}
+
+dependencies {
+    if (configurations.find { it.name == 'documentation' }) {
+        documentation 'info.picocli:picocli:3.8.0'
+        documentation "org.fusesource.jansi:jansi:1.18"
+        documentation "jline:jline:2.14.6"
+        documentation "com.github.javaparser:javaparser-core:3.15.14"
+        documentation "org.apache.groovy:groovy:${groovyVersion}"
+        documentation "org.apache.groovy:groovy-templates:${groovyVersion}"
+        documentation "org.apache.groovy:groovy-ant:$groovyVersion"
+        documentation "org.apache.groovy:groovy-cli-picocli:$groovyVersion"
+    }
+}
+
+def cleanTask = project.tasks.findByName("clean")
+if (cleanTask == null) {
+    task clean(type: Delete) {
+        delete(buildDir)
+    }
+} else {
+    cleanTask.doLast {
+        ant.delete(dir: "build/docs")
+    }
+}
+
+tasks.withType(Groovydoc) {
+    group = DOCUMENTATION_GROUP
+    docTitle = "${project.title} - ${project.version}"
+    destinationDir = project.file("build/docs/api")
+    def files = []
+    project.rootProject.subprojects
+            .findAll { !it.name != 'docs' && !it.name.startsWith('examples') }
```

Comment:
`!it.name != 'docs'` does not seem right

Reply:
Updated.  Certainly an advantage of removing the "invisible" commonBuild approach.  

matches the way it is done in gorm-hibernate5
https://github.com/grails/gorm-hibernate5/blob/0f3e6ed84a3c552de023c4ec278bc663548f4bf1/docs/build.gradle#L48